src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/AnalysisDefaults.scala

```diff
@@ -0,0 +1,62 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Location, Tag}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.LanguageDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.LocationsExtractor
+import com.microsoft.partnercatalyst.fortis.spark.transforms.people.PeopleRecognizer
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetector
+import com.microsoft.partnercatalyst.fortis.spark.transforms.topic.KeywordExtractor
+
+/**
+  * Provides default analysis method implementations for a concrete [[Analyzer]].
+  *
+  * By implementing these traits, the concrete [[Analyzer]] explicitly enables the default implementations that are
+  * applicable to it. If a default is not enabled, the compiler will enforce that a custom implementation is provided.
+  *
+  */
+private[analyzer] object AnalysisDefaults {
```

Comment:
I'd considered that, but I believe default analysis methods make more sense as an opt-in feature since it's possible that they aren't applicable for every pipeline. It seems like a code smell to me for N.A. methods to be lurking in a base class, ready for someone to forget they're there.

Really this is just syntactic sugar for calling a shared helper method explicitly for each concrete method implementation. If an Analyzer implements a particular default trait, then the corresponding analysis method implementation will be present in the base class (and defaulted to if the concrete Analyzer does not override it, or callable with super() if it does). 

In other words, a default analysis implementation will be available in the base class hierarchy of a concrete `Analyzer` if and only if the concrete Analyzer explicitly denotes that a default implementation is applicable to it by implementing the trait.